hack/run-hive-locally.sh

```diff
@@ -0,0 +1,365 @@
+On pressing ctrl + c, the script is terminated and relevant components are scaled back up.
+
+Warnings:
+- The active cluster may be in use, in which case the script could interfere with its functioning.
+
+Notes:
+- Artifacts (ConfigMaps containing in-cluster Hive configurations and log files) produced are stored in
+the --artifacts_folder which is removed once the script ends unless --preserve-artifacts is specified.
+- Selecting $hive_component_controllers runs all Hive controllers (including hive-clustersync) together
+as a single replica.
+- Endpoints of the Hive component running locally (e.g. metrics, pprof) are directly exposed on localhost.
+
+Usage:
+- ./hack/run-hive-locally.sh COMPONENT [--flag=value ...]
+
+Parameters:
+- COMPONENT: must be either $hive_component_operator or $hive_component_controllers
+
+Optional flags:
+- --hive-operator-ns=NAMESPACE, where NAMESPACE defaults to hive
+- --hive-operator-log-level=LOG_LEVEL, where LOG_LEVEL defaults to debug
+- --hive-controllers-log-level=LOG_LEVEL, where LOG_LEVEL defaults to debug
+- --artifacts-folder=DIRECTORY, where DIRECTORY defaults to artifacts
+- --preserve-artifacts
+- --help"
+}
+
+function kill_background_jobs {
+    local pid
+    while read -r pid; do
+        hl_echo "Killing job $pid"
+        kill "$pid" || true
+        wait "$pid" || true
+    done < <(jobs -p)
+}
+
+function get_envs {
+    local type_name="$1"
+    local ns="$2"
+    # Arrange output into the following format:
+    # name1:value1
+    # name2:value2
+    # ...
+    local jsonpath_query='{range .spec.template.spec.containers[0].env[*]}{@.name}={@.value}{"\n"}{end}'
+    local env_value
+    local env
+    local value
+    local resolved_value
+
+    hl_echo "Getting environment variables from $type_name in ns/$ns"
+    while read -r env_value; do
+        env="${env_value%=*}"
+
+        # Skip environment variables that are already resolved
+        if [ ${resolved_envs["$env"]+_} ]; then
+            hl_echo "Environment variable $env already resolved, skipping"
+            continue
+        fi
+
+        # Skip irrelevant environment variables
+        if [[ $env = "CLI_CACHE_DIR" ]] || [[ $env = "HIVE_SKIP_LEADER_ELECTION" ]]; then
+            hl_echo "Environment variable $env is irrelevant, skipping"
+            continue
+        fi
+
+        # Resolve the environment variable's value
+        value="${env_value#*=}"
+        case "$value" in
+        "")
+            hl_echo "Environment variable $env does not have a .value, skipping"
+            continue
+            ;;
+        /*)
+            resolved_value="./$artifacts_folder/$(basename "$value")-${hc_spec_generation}.txt"
+            # Skip if the file does not exist
+            if ! oc exec "$type_name" -n "$ns" -- /bin/bash -c "cat $value" > "$resolved_value"; then
+                rm "$resolved_value"
+                continue
```

Comment:
There are other reasons this `oc exec` could fail, and I would think we want to blow up on most of them. Also, again it is technically possible that "file exists and is empty" behaves differently than "file does not exist". This isn't beautiful, but...
```suggestion
            oc exec "$type_name" -n "$ns" -- /bin/bash -c "[ -f $value ] && cat $value || echo -n '!ENOENT!'" > "$resolved_value"
            if [ '!ENOENT!" == "$(cat $resolved_value)" ]; then
                rm "$resolved_value"
                continue
```

Reply:
Thanks for the suggestion. Fixed.